### ISSUE TITLE
rxjs-no-redundant-notify is too simplistic

### ISSUE THREAD

**Issue by kamocuvao** [cartant/rxjs-tslint-rules#108]
First of all, thank you for providing these linters, I have already found some bugs with them.

Now for this Issue, I found that `rxjs-no-redundant-notify` only checks, if no `error` `complete` or `next` follows `error` or `complete`.

It does not however check the origin objects, where these functions come from, even if its from totally different classes (like a logger). It also does not check for scoping.

Here is a minimal example:
```ts
import { Subject } from 'rxjs';

export class A {
  private aSubject$ = new Subject();
  private bSubject$ = new Subject();
  
  doesNotWork(){
    this.aSubject$.error("err");
    this.bSubject$.error("err"); //<--- Redundant notifications are forbidden
  }

  doesNotWorkToo(){
    Logger.error("err");
    this.bSubject$.error("err"); //<--- Redundant notifications are forbidden
  }

  doesAlsoNotWork(){
    Logger.error("err");
    Logger.error("err"); //<--- Redundant notifications are forbidden
  }

  doesWorkButShouldNot(){
    if(Logger.isLoggable()){
      this.aSubject$.error("err");
    }
    this.aSubject$.error("err3");
  }
}

export class Logger {
  public static error(error: string){
    console.log(error);
  }

  public static isLoggable(){
    return true;
  }
}
```

This makes this rule unusable as it is now.

**Comment by cartant** [comment]
Yeah, checking the base identifier and its type is definitely something that should be done, but not catching things like this is by design:

```ts
  doesWorkButShouldNot(){
    if(Logger.isLoggable()){
      this.aSubject$.error("err");
    }
    this.aSubject$.error("err3");
  }
```

The calls have to be immediate siblings. Anything other than that is too much for static analysis, IMO.

**Comment by cartant** [comment]
The false positives should be fixed in 4.26.2.

**Comment by kamocuvao** [comment]
Thank you, the examples I posted work now. Still I have one false positive left:

```ts
import { Subject } from 'rxjs';

export class A {
  private aSubject$ = new Subject();
  private bSubject$ = new Subject();

  doesWork() {
    this.aSubject$.complete();
    this.bSubject$.complete();
  }

  doesNotWork() {
    this.aSubject$.complete();
    this.bSubject$.next();
    this.bSubject$.complete(); //<--- Redundant notifications are forbidden
  }
}

```

**Comment by cartant** [comment]
Fixed in `4.26.3`.
